### PR TITLE
Improve accumulated power calculation

### PR DESCRIPTION
**Pull request by eunsoo-icon**
- calculate accumulated power of PRep with voted and bonded whenever voted and bonded values change
- remove PRep.accumulatedBonded

### CODE REVIEW

File: icon/iiss/calculator/prep.go [hunk 1]
```diff
@@ -366,27 +352,26 @@ func (p *PRepInfo) InitAccumulated() {
...
-			continue
-		} else {
-			prep.ApplyVote(vType, vote.Amount(), p.offsetLimit-offset)
-			p.log.Debugf("ApplyVote %+v: by %d, %d %+v, %d * %d",
-				prep, vType, offset, vote, vote.Amount(), p.offsetLimit-offset)
-			p.preps[key] = prep
+		prep, ok := p.preps[key]
+		if !ok {
+			prep = p.Add(vote.To(), icmodule.ESDisablePermanent, new(big.Int), new(big.Int), 0, false)
 		}
```
Comment (mksong76) [inline icon/iiss/calculator/prep.go:358]: Previous code skip it, but currently it adds one.
Does it fix the other problem? or just invalid fixes?

**Review by mksong76 — COMMENTED** [review]
Please confirm the logic changes in prep.go

File: icon/iiss/calculator/prep.go [hunk 2]
```diff
@@ -366,27 +352,26 @@ func (p *PRepInfo) InitAccumulated() {
...
-		if prep, ok := p.preps[key]; !ok {
-			continue
-		} else {
-			prep.ApplyVote(vType, vote.Amount(), p.offsetLimit-offset)
-			p.log.Debugf("ApplyVote %+v: by %d, %d %+v, %d * %d",
-				prep, vType, offset, vote, vote.Amount(), p.offsetLimit-offset)
-			p.preps[key] = prep
+		prep, ok := p.preps[key]
+		if !ok {
+			prep = p.Add(vote.To(), icmodule.ESDisablePermanent, new(big.Int), new(big.Int), 0, false)
```
Comment (mksong76) [inline icon/iiss/calculator/prep.go:357]: Logic is changed to add PRep structure if it doesn't exist

File: icon/iiss/calculator/prep_test.go [hunk 3]
```diff
@@ -41,7 +41,8 @@ type prep struct {
 }
 
 func newTestPRep(p prep) *PRep {
-	return NewPRep(p.owner, p.status, big.NewInt(p.delegate), big.NewInt(p.bond), p.commissionRate, p.pubkey)
+	tp := NewPRep(p.owner, p.status, big.NewInt(p.delegate), big.NewInt(p.bond), p.commissionRate, p.pubkey)
+	return tp
```
Comment (mksong76) [inline icon/iiss/calculator/prep_test.go:45]: Unnecessary splitting

File: icon/iiss/calculator/prep.go (hunk 2 above)
Comment (eunsoo-icon) [reply icon/iiss/calculator/prep.go:357]: Previous code had a bug.

This code works when ICONist delegate to not registered P-Rep.

If ICONist delegate to not registered P-Rep, they can't get rewards. After that P-Rep register, they must get rewards with past delegation. So we must store information of that P-Rep.

**Review by mksong76 — APPROVED** [review]

**Pull request merged**